packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -298,26 +299,35 @@ Plugin _pluginFromPackage(String name, Uri packageRoot) {
   );
 }
 
-List<Plugin> findPlugins(FlutterProject project) {
+Future<List<Plugin>> findPlugins(FlutterProject project) async {
   final List<Plugin> plugins = <Plugin>[];
-  Map<String, Uri> packages;
+  PackageConfig packageConfig;
   try {
     final String packagesFile = globals.fs.path.join(
       project.directory.path,
-      PackageMap.globalPackagesPath,
+      globalPackagesPath,
+    );
+    packageConfig = await loadPackageConfigUri(
+      globals.fs.file(packagesFile).absolute.uri,
+      loader: (Uri uri) {
+        final File file = globals.fs.file(uri);
+        if (!file.existsSync()) {
+          return null;
+        }
+        return file.readAsBytes();
```

Comment:
Here and else where, can we change these to a synchronous read to hopefully make any crash stack traces easier to read? Like:
```dart
...
final Uint8List data = file.readAsBytesSync();
return Future<Uint8List>.value(data);
```